packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -89,10 +92,40 @@ export async function streamExtractExternal(
         clearTimeout(timer)
         reject(error)
       })
-      .pipe(unzip.Extract({path: directory}))
-      .on('close', () => {
+      .pipe(unzip.Parse())
+      .on('entry', (entry: unzip.Entry) => {
```

Comment:
Wondering whether we need any new tests for the new logic or whether we feel confident that we have coverage based on existing tests?

Reply:
Added a test, however the current tests were very helpful for validating this change so I'm pretty confident in the existing ones!